### PR TITLE
feat(specification-storage): implement PUT/DELETE endpoints for subfields

### PR DESCRIPTION
**Pull request by psmagin**
### Purpose
implement PUT/DELETE endpoints for subfields

### Approach
- define endpoints
- implement endpoints logic
- update compose to support local dev testing

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [ ] **Database Schema Changes**: Indicate any database schema changes and their impact. Confirm that migration scripts were created.
- [ ] **Interface Version Changes**: Indicate any changes to interface versions.
- [ ] **Interface Dependencies**: Document added or removed dependencies.
- [x] **Permissions**: Document any changes to permissions.
- [x] **Logging**: Confirm that logging is appropriately handled.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **Manual Testing**: Confirm that changes were tested on local or dev environment.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
https://folio-org.atlassian.net/browse/MRSPECS-24

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -268,6 +286,16 @@
...
     },
+    {
+      "permissionName": "specification-storage.subfields.item.put",
+      "displayName": "Specification Storage - Update indicator code definition",
+      "description": "Update indicator code definition"
+    },
+    {
+      "permissionName": "specification-storage.subfields.item.delete",
+      "displayName": "Specification Storage - Delete indicator code definition",
+      "description": "Delete indicator code definition"
```
Comment (viacheslavkol) [inline descriptors/ModuleDescriptor-template.json:297]: subfield definition

File: mod-record-specifications-server/src/main/java/org/folio/rspec/service/SubfieldService.java [hunk 2]
```diff
@@ -1,39 +1,86 @@
...
 @RequiredArgsConstructor
 public class SubfieldService {
 
-  private final SubfieldRepository subfieldRepository;
-  private final SubfieldMapper subfieldMapper;
+  private final SubfieldRepository repository;
+  private final SubfieldMapper mapper;
+  private final EventProducer<UUID, SpecificationUpdatedEvent> eventProducer;
+
+  private final Map<Scope, ScopeValidator<SubfieldChangeDto, Subfield>> fieldValidators = new EnumMap<>(Scope.class);
```
Comment (viacheslavkol) [inline mod-record-specifications-server/src/main/java/org/folio/rspec/service/SubfieldService.java:36]:
```suggestion
  private final Map<Scope, ScopeValidator<SubfieldChangeDto, Subfield>> scopeValidators = new EnumMap<>(Scope.class);
```

Comment (viacheslavkol) [reply mod-record-specifications-server/src/main/java/org/folio/rspec/service/SubfieldService.java:36]: Or `subfieldValidators`/`validators`

File: mod-record-specifications-server/src/test/java/org/folio/rspec/controller/SpecificationStorageSubfieldsControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import org.springframework.test.web.servlet.MockMvc;
+
+@UnitTest
+@ExtendWith(RandomParametersExtension.class)
+@WebMvcTest(SpecificationStorageSubfieldsController.class)
+@Import({ApiExceptionHandler.class, TranslationConfig.class, ValidationConfig.class})
+@ComponentScan(basePackages = {"org.folio.rspec.controller.handler",
+                               "org.folio.rspec.service.i18n",
+                               "org.folio.spring.i18n"})
+class SpecificationStorageSubfieldsControllerTest {
```
Comment (viacheslavkol) [inline mod-record-specifications-server/src/test/java/org/folio/rspec/controller/SpecificationStorageSubfieldsControllerTest.java:47]: Would be good to have also basic Integration Test cases

**Review by viacheslavkol — CHANGES_REQUESTED** [review]

**Review by mukhiddin-yusuf — APPROVED** [review]

**Review by viacheslavpoliakov — APPROVED** [review]

**Pull request merged**